### PR TITLE
feat:[CDS-70050]: Serverless Prepare Rollback Container Changes

### PR DESCRIPTION
**Pull request by piyush-bhuwalka**
## Describe your changes

Serverless Prepare Rollback Container Changes for Manager Side are done as part of this PR. Additionally we have plugin info provider needer for container steps

## Checklist
- [x] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48850)

### CODE REVIEW

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger checkstylecodeformat

**Comment by piyush-bhuwalka** [comment]
trigger messagecheck

**Comment by piyush-bhuwalka** [comment]
trigger messagecheck

**Comment by piyush-bhuwalka** [comment]
trigger pmd

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger messagecheck

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Comment by piyush-bhuwalka** [comment]
trigger uts

File: 125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/steps/serverless/ServerlessAwsLambdaPrepareRollbackContainerStepPlanCreator.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
+ */
+
+package io.harness.cdng.creator.plan.steps.serverless;
+
+import static io.harness.cdng.visitor.YamlTypes.DOWNLOAD_MANIFESTS;
```
Comment (sainath-b) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/steps/serverless/ServerlessAwsLambdaPrepareRollbackContainerStepPlanCreator.java:10]: Create a new yaml type for Download Manifests Serverless V2

File: 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaPrepareRollbackDataOutcome.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import lombok.Builder;
+import lombok.Value;
+import org.springframework.data.annotation.TypeAlias;
+
+@OwnedBy(HarnessTeam.CDP)
+@Value
+@Builder
+@TypeAlias("serverlessAwsLambdaRollbackDataOutcome")
+@JsonTypeName("serverlessAwsLambdaRollbackDataOutcome")
```
Comment (sainath-b) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaPrepareRollbackDataOutcome.java:26]: Change this to preparRollbackDataOutcome

File: 125-cd-nextgen/src/test/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaPrepareRollbackContainerStepTest.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    ServerlessAwsLambdaPrepareRollbackContainerStepParameters stepParameters =
+        ServerlessAwsLambdaPrepareRollbackContainerStepParameters.infoBuilder()
+            .image(ParameterField.<String>builder().value("sdaf").build())
+            .build();
+    StepElementParameters stepElementParameters = StepElementParameters.builder().spec(stepParameters).build();
+
+    String logKey = "logKey";
+    long timeout = 1000;
+    String parkedTaskId = "parkedTaskId";
+    assertThat(serverlessAwsLambdaPrepareRollbackContainerStep.getSerialisedStep(
```
Comment (sainath-b) [inline 125-cd-nextgen/src/test/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaPrepareRollbackContainerStepTest.java:125]: add field level assertions

File: 127-cd-nextgen-entities/src/main/java/io/harness/executions/steps/StepSpecTypeConstants.java [hunk 4]
```diff
@@ -126,8 +126,11 @@ public interface StepSpecTypeConstants {
   String AWS_SAM_DEPLOY = "AwsSamDeploy";
   String AWS_SAM_BUILD = "AwsSamBuild";
   String AWS_SAM_ROLLBACK = "AwsSamRollback";
+  String DOWNLOAD_MANIFESTS = "DownloadManifests";
```
Comment (sainath-b) [inline 127-cd-nextgen-entities/src/main/java/io/harness/executions/steps/StepSpecTypeConstants.java:129]: use download manifests serverless

File: 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/ServerlessStepCommonHelper.java [hunk 5]
```diff
@@ -635,4 +655,34 @@ private ConnectorInfoDTO getConnectorDTO(String connectorId, Ambiance ambiance)
...
     return serverlessEntityHelper.getConnectorInfoDTO(connectorId, ngAccess);
   }
+
+  public void verifyPluginImageIsProvider(ParameterField<String> image) {
+    if (ParameterField.isNull(image) || image.getValue() == null) {
+      throw new InvalidRequestException("Plugin Image must be provided");
+    }
+  }
+
+  public void putValuesYamlEnvVars(Ambiance ambiance,
```
Comment (sainath-b) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/ServerlessStepCommonHelper.java:665]: Can you remove these methods if not required

File: 125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/steps/serverless/ServerlessAwsLambdaPrepareRollbackContainerStepPlanCreator.java (hunk 1 above)
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/steps/serverless/ServerlessAwsLambdaPrepareRollbackContainerStepPlanCreator.java]: Done

File: 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaPrepareRollbackDataOutcome.java (hunk 2 above)
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaPrepareRollbackDataOutcome.java]: Done

File: 125-cd-nextgen/src/test/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaPrepareRollbackContainerStepTest.java (hunk 3 above)
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/test/java/io/harness/cdng/serverless/container/steps/ServerlessAwsLambdaPrepareRollbackContainerStepTest.java]: Done

File: 127-cd-nextgen-entities/src/main/java/io/harness/executions/steps/StepSpecTypeConstants.java (hunk 4 above)
Comment (piyush-bhuwalka) [reply 127-cd-nextgen-entities/src/main/java/io/harness/executions/steps/StepSpecTypeConstants.java]: Done

File: 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/ServerlessStepCommonHelper.java (hunk 5 above)
Comment (piyush-bhuwalka) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/serverless/ServerlessStepCommonHelper.java:665]: We do not need to remove

**Comment by piyush-bhuwalka** [comment]
trigger uts

**Review by sainath-b — DISMISSED** [review]

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 5 of 36 files at r1, 4 of 14 files at r3, 18 of 26 files at r4, 1 of 1 files at r5, 1 of 3 files at r6, 2 of 4 files at r7, 5 of 5 files at r8, 3 of 5 files at r9, 5 of 5 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48850)* status: 44 file(s) to review. 5. unresolved discussions. 1 of 2 approvals obtained.

**Review by sainath-b — APPROVED** [review]

**Comment by piyush-bhuwalka** [comment]
trigger review

**Pull request merged**